metrics/collector.go

```diff
@@ -0,0 +1,64 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rs/zerolog"
+)
+
+type Collector interface {
+	ApiErrorOccurred()
+	MeasureRequestDuration(start time.Time, labels prometheus.Labels)
+}
+
+type DefaultCollector struct {
+	// TODO: for now we cannot differentiate which api request failed number of times
+	apiErrorsCounter prometheus.Counter
+	requestDurations *prometheus.HistogramVec
+}
+
+func NewCollector(logger zerolog.Logger) Collector {
+	apiErrors := prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "api_errors_total",
+		Help: "Total number of errors returned by the endpoint resolvers",
+	})
+
+	// TODO: Think of adding 'status_code'
+	requestDurations := prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "api_request_duration_seconds",
```

Comment:
make it `ms` instead of `s`

Reply:
The suffix `_seconds` is just a Prometheus naming convention. We'll see this value in ms/ns/whatever when scrap the metrics with a backend